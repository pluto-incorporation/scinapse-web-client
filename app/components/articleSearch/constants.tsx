export const LOCAL_STORAGE_FILTERS = "l_s_filters";
export const PREVIOUS_FILTER = "prev_filters";
