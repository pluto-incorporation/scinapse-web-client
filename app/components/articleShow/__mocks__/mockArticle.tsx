export function getMockArticle() {
  return {
    articleId: 1,
    tags: ["Open Access Paper", "CAU Paper"],
  };
}
