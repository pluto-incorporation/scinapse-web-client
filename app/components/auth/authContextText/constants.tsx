export const positiveSignUpContext: { [key: string]: string } = {
  downloadPdf: "📜 To download PDF more, you need to be a Scinapse member.",
  citePaper: "🔍 To copy citation, you need to be a Scinapse member.",
  viewMorePDF: "📜 To view full text, you need to be a Scinapse member.",
  viewMorePDFUnlimited: "📜 To view full text, you need to be a Scinapse member.",
  addToCollection: "🔍 To add papers to collection, you need to be a Scinapse member.",
  clickRequestFullTextBtn: "📜 To request the full text of this paper, you need to be a Scinapse member.",
  query: "🔍 To search more, you need to be a Scinapse member.",
  paperShow: "📈 To view paper information more, you need to be a Scinapse member.",
  paperFromSearch: "📜 To view more papers, you need to be a Scinapse member.",
  queryLover: "💌 Hey Scinapse lover! Please help us to fully support your research by signing up.",
  nextPageFromSearch: "➡ To go to the next page, you need to be a Scinapse member.",
};
