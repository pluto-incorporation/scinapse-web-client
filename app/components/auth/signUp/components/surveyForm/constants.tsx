interface CheckedAnswer {
  name: string;
  order: number;
}

export interface RawQuestionType {
  question: string;
  answers: string[];
  random: boolean;
  type: string;
  description?: string;
}

interface QuestionType {
  question: string;
  random: boolean;
  checked: CheckedAnswer[];
}

export interface RawSurvey {
  surveyName: string;
  questions: RawQuestionType[];
}

export interface Survey {
  surveyName: string;
  questions: QuestionType[];
}
export const SCINAPSE_SURVEY_NAME = 'thinkAsScinapse';

export const SCINAPSE_SURVEY_QUESTIONS: RawSurvey = {
  surveyName: SCINAPSE_SURVEY_NAME,
  questions: [
    {
      question: 'What do you think as Scinapse?',
      answers: ['Publisher', 'PDF Provider', 'Search Engine'],
      random: true,
      type: 'radio',
    },
    { question: 'alpha', answers: ['a', 'b', 'c'], random: true, type: 'checkbox' },
  ],
};
