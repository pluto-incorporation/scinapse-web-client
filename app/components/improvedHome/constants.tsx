export const JOURNALS = [
  'nature',
  'science',
  'ieee',
  'cell',
  'acs',
  'aps',
  'lancet',
  'acm',
  'jama',
  'bmj',
  'pnas',
  'more-journals',
];

export const MOBILE_JOURNALS = ['nature', 'science', 'lancet', 'acm', 'ieee', 'cell', 'more-journal-mobile'];
