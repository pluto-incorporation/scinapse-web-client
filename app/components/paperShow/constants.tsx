export type RELATED_PAPERS = 'reference' | 'cited';
