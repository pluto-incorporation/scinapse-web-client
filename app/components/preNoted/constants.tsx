interface BlockedBubbleKeywords {
  verb: string;
  noun: string;
}

interface BlockedBubbleContextStructure {
  title: string;
  mainText: string;
}

const blockedBubbleVerbAndNoun: { [key: string]: BlockedBubbleKeywords } = {
  downloadPdf: { verb: "download", noun: "PDF" },
  clickRequestFullTextBtn: { verb: "request", noun: "paper" },
  viewMorePDF: { verb: "view", noun: "PDF" },
  addToCollection: { verb: "save to", noun: "Collection" },
};

export function getBlockedBubbleContext(userGroupName: string, buttonAction: string): BlockedBubbleContextStructure {
  const contextVerbAndNoun = blockedBubbleVerbAndNoun[buttonAction];
  switch (userGroupName) {
    case "1":
      return {
        title: `Do you want to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}?`,
        mainText: `Join Scinapse to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}. It's free!`,
      };

    case "2":
      return {
        title: `Join Scinapse to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}`,
        mainText: `When you join Scinapse, you can ${contextVerbAndNoun.verb} any ${contextVerbAndNoun.noun}.`,
      };

    case "3":
      const capitalizedVerb = contextVerbAndNoun.verb.charAt(0).toUpperCase + contextVerbAndNoun.verb.slice(1);
      return {
        title: `${capitalizedVerb} ${contextVerbAndNoun.noun} with a Scinapse account`,
        mainText: `Enjoy unlimited Scinapse with a Scinapse account.`,
      };
  }
  return { title: "", mainText: "" };
}
