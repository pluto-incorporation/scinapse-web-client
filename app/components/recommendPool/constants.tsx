export const BASED_ACTIVITY_COUNT_STORE_KEY = 'recommendPapersDialogCount';
export const ALREADY_VISITED_RECOMMEND_PAPERS = 'visited';
export const BASED_ACTIVITY_PAPER_IDS_FOR_NON_USER_KEY = 'b_a_p_ids';
