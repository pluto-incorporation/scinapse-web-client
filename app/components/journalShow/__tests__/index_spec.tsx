import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter, Route } from 'react-router';
import { Provider } from 'react-redux';
import { generateMockStore } from '../../../__tests__/mockStore';
import { initialState } from '../../../reducers';
import JournalShowContainer from '..';
import { RAW } from '../../../__mocks__';
import { JOURNAL_SHOW_PATH } from '../../../constants/routes';

jest.mock('../../../api/journal');

describe('JournalShow Container Component', () => {
  let mockStore = generateMockStore(initialState);

  beforeEach(() => {
    mockStore.clearActions();
  });

  describe('when journal data exist', () => {
    beforeEach(async () => {
      const journalPaper = RAW.JOURNAL_PAPERS_RESPONSE.data.content[0];

      const mockState = {
        ...initialState,
        journalShow: {
          ...initialState.journalShow,
          journalId: 2764552960,
          paperCurrentPage: 1,
          paperIds: ['8107'],
          paperTotalPage: 1,
        },
        entities: {
          ...initialState.entities,
          journals: { [`${RAW.JOURNAL.id}`]: RAW.JOURNAL },
          papers: { [`${journalPaper.id}`]: journalPaper },
        },
      };
      mockStore = generateMockStore(mockState);
    });

    it.skip('should render correctly', () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <MemoryRouter initialIndex={0} initialEntries={['/journals/2764552960']}>
              <Route path={JOURNAL_SHOW_PATH}>
                {/*Route is needed for providing match params*/}
                <JournalShowContainer />
              </Route>
            </MemoryRouter>
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
