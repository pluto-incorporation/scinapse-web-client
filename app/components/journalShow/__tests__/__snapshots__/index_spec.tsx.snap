// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalShow Container Component when journal data exist should render correctly 1`] = `
<div
  className="journalShowWrapper"
>
  <div
    className="headSection"
  >
    <div
      className="container"
    >
      <div
        className="leftBox"
      >
        <div
          className="title"
        >
          <a
            href="/journals/2764552960"
            onClick={[Function]}
          >
            Science & Engineering Faculty
          </a>
        </div>
        <div
          className="infoWrapper"
        >
          <span>
            <div
              className="subtitle"
            >
              Papers
            </div>
            <strong>
              8859
            </strong>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="paperListContainer"
  >
    <div
      className="container"
    >
      <div
        className="leftBox"
      >
        <div
          className="paperListBox"
        >
          <div
            className="header"
          >
            <div
              className="listTitle"
            >
              <span>
                Papers 
              </span>
              <span
                className="paperCount"
              >
                0
              </span>
            </div>
            <div
              className="searchInputWrapper"
            >
              <div
                className="inputBox"
                style={undefined}
              >
                <input
                  autoFocus={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search papers in this journal"
                  style={undefined}
                  value=""
                />
                <div
                  className="icon"
                  onClick={[Function]}
                >
                  <i
                    className="icon"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="subHeader"
          >
            <div
              className="resultPaperCount"
            >
              1 page of 1 pages (0 results)
            </div>
            <div
              className="sortBoxWrapper"
            >
              <div
                className="sortBoxWrapper"
              >
                <div
                  className="currentOption"
                  onClick={[Function]}
                >
                  <span
                    className="sortOptionText"
                  >
                    Newest
                  </span>
                  <i
                    className="icon downArrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="paperItemWrapper"
            >
              <div
                className="contentSection"
              >
                <div>
                  <a
                    className="title"
                    href="/papers/8107"
                    onClick={[Function]}
                  >
                    <span>
                      Understanding the very nature of Project Management : a praxiological approach
                    </span>
                  </a>
                  <a
                    className="newTabIconWrapper"
                    href="/papers/8107"
                    onClick={[Function]}
                    target="_blank"
                  >
                    <i
                      className="icon newTabIcon"
                    />
                  </a>
                </div>
                <div
                  className="publishInfoList"
                >
                  <div
                    className="journal"
                  >
                    <i
                      className="icon"
                    />
                    <div
                      className="journalText"
                    >
                      <span
                        className="bold"
                      >
                        2005
                         in 
                      </span>
                      <a
                        className="journalName"
                        href="/journals/2764552960"
                        onClick={[Function]}
                      >
                        Science & Engineering Faculty
                      </a>
                    </div>
                  </div>
                  <div
                    className="author"
                  >
                    <i
                      className="icon"
                    />
                    <span
                      className="authors"
                    >
                      <span
                        className="author"
                      >
                        <a
                          className="authorName"
                          href="/authors/538589259"
                          onClick={[Function]}
                        >
                          Christophe Bredillet
                        </a>
                        <span
                          className="authorHIndex"
                        >
                          <span
                            className="hIndexChar"
                          >
                            13
                          </span>
                          <div
                            className="tooltipContainer authorHIndexTooltip"
                            style={
                              Object {
                                "left": -55,
                                "top": -42,
                              }
                            }
                          >
                            <div
                              className="innerContainer"
                            >
                              <div
                                className="tooltipIconWrapper"
                                style={
                                  Object {
                                    "top": -9,
                                  }
                                }
                              >
                                <i
                                  className="icon"
                                />
                              </div>
                              <div
                                className="tooltipContent"
                              >
                                Estimated H-index: 13
                              </div>
                            </div>
                          </div>
                        </span>
                         (Queensland University of Technology)
                      </span>
                    </span>
                  </div>
                </div>
                <span
                  className="abstract"
                >
                  Foreword: In this paper I call upon a praxiological approach. Praxeology (early alteration of praxiology) is the study of human action and conduct. The name praxeology/praxiologyakes is root in praxis, Medieval Latin, from Greek, doing, action, from prassein to do, practice (Merriam-Webster Dictionary). Having been involved in project management education, research and practice for the last twenty...
                </span>
                <div
                  className="infoList"
                >
                  <a
                    className="citedButton"
                    href="/papers/8107#cited"
                    onClick={[Function]}
                  >
                    <span>
                      Cited 6
                    </span>
                  </a>
                  <a
                    className="sourceButton"
                    href="http://eprints.qut.edu.au/49501/"
                    onClick={[Function]}
                    target="_blank"
                  >
                    <i
                      className="icon sourceButtonIcon"
                    />
                    <span>
                      Source
                    </span>
                  </a>
                  <button
                    className="addCollectionBtnWrapper"
                    onClick={[Function]}
                  >
                    <i
                      className="icon plusIcon"
                    />
                    <span>
                      Add Collection
                    </span>
                  </button>
                  <div
                    className="claimButton"
                  >
                    <div>
                      <button
                        className="MuiButtonBase-root-10 MuiIconButton-root-4 additionalMenuIcon"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex="0"
                        type="button"
                      >
                        <span
                          className="MuiIconButton-label-9"
                        >
                          <i
                            className="icon ellipsisIcon"
                          />
                        </span>
                        <span
                          className="MuiTouchRipple-root-13"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="paginationWrapper"
              style={
                Object {
                  "margin": "24px 0",
                }
              }
            >
              <div
                className="prevButtons"
              />
              <span
                className="pageItem active"
                onClick={[Function]}
                style={undefined}
              >
                1
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rightBox"
      >
        <div
          className="fosSection"
        >
          <div
            className="topFosTitle"
          >
            Top fields of study
          </div>
          <div
            className="fosWrapper"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footerContainer"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="title"
    >
      <div>
        © 2018 Pluto Network. All rights reserved
      </div>
    </div>
    <div
      className="rightBox"
    >
      <a
        className="link"
        href="https://www.facebook.com/PlutoNetwork/"
        onClick={[Function]}
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="link"
        href="https://twitter.com/pluto_network"
        onClick={[Function]}
        target="_blank"
      >
        Twitter
      </a>
      <a
        className="link"
        href="https://medium.com/pluto-network"
        onClick={[Function]}
        target="_blank"
      >
        Blog
      </a>
      <a
        className="link"
        href="https://pluto.network"
        onClick={[Function]}
        target="_blank"
      >
        About us
      </a>
      <a
        className="link"
        href="mailto:team@pluto.network"
        onClick={[Function]}
        target="_blank"
      >
        Contact us
      </a>
      <a
        className="termsOfServiceLink"
        href="/terms-of-service"
        onClick={[Function]}
      >
        Terms of service
      </a>
    </div>
  </footer>
</div>
`;
